src/handwritten/stream_readers/analog_multi_channel_reader.py

```diff
@@ -211,12 +211,15 @@ def read_waveforms(
         
         for i, waveform in enumerate(waveforms):
             if number_of_samples_per_channel > waveform.sample_count:
```

Comment:
Users may hit this when reusing a waveform after a short read. It would be better to limit based on capacity.